libminifi/include/core/ContentRepository.h

```diff
@@ -62,15 +73,24 @@ class ContentRepository : public core::CoreComponent, public StreamManager<minif
     return getName();
   }
 
+  std::unique_ptr<StreamAppendLock> append(const ResourceClaim& claim, size_t offset) override;
```

Comment:
Shouldn't this return a `ContentStreamAppendLock`?

Reply:
I don't think we can override with a different return type even if they are convertible

Reply:
You're right, it only works with pointers and references, but not `unique_ptr`.